qiskit_sphinx_theme/furo/base/static/styles/qiskit_changes.css

```diff
@@ -30,6 +30,24 @@ body {
   top: var(--qiskit-top-nav-bar-height);
 }
 
+/* Fix # anchor links not accounting for the top nav bar.
+*
+* The rules for mobile come from Furo, but we add --qiskit-top-nav-bar-height.
+* But for some reason, the default rule has too much spacing if we include Furo's original rule's
+* --header-height. */
+:target {
```

Comment:
TIL from Furo's code that `:target` matches every HTML element that corresponds to the `#` in the URL. FYI those hashes are based on the HTML `id`, e.g. `<section id="electron">`.